### ISSUE TITLE
no-cycle rule not working with babel-eslint parser

### ISSUE THREAD

**Issue by jay1337** [benmosher/eslint-plugin-import#1166]
Here is a simple repro:

package.json
```
{
  "dependencies": {
    "babel-eslint": "^9.0.0",
    "eslint": "^5.4.0",
    "eslint-plugin-import": "^2.14.0"
  }
}
```
.eslintrc
```
{
	parser: "babel-eslint",
	parserOptions: {
		sourceType: 'module',
	},
	plugins: ['import'],
	rules: { 'import/no-cycle': 'error' },
}
```

a.js
```
import B from './b';
export default "A" + B;
```

b.js
```
import A from './a';
export default A + "B";
```

No error is found by eslint but if default parser is used (by removing line `parser: "babel-eslint",`) then `no-cycle` errors are found in each file.

Is it the expected behaviour ? An issue with `eslint-plugin-import` ? An issue with `babel-eslint` ?

**Comment by gajus** [comment]
> If I comment that return out, the the no-cycle errors are reported as expected.

It was added for support of ESLint <5.

**Comment by edmorley** [comment]
Ok, but the rule is now being skipped for ESLint 5 too. Should we just version-check, or can you remember more specifically what was missing, so that we can feature detect that instead?

**Comment by gajus** [comment]
Do not remember more specifically.

**Comment by bora89** [comment]
Hey guys, any progress on this? We experience the same issue, with babel parser no-cycle does not work
